justfile

```diff
@@ -420,18 +420,6 @@ rustdoc-serve:
 
 # Build for each separate commit (for "pull_request") or for the HEAD of the branch (other events)
 [script]
-build-sweep start="main":
+build-sweep start="main" command="{ git log --oneline --no-decorate -n 1 && just cargo build; }":
     {{ _just_debuggable_ }}
-    set -euo pipefail
-    if [ {{ _clean }} != "clean" ]; then
-      >&2 echo "can not build-sweep with dirty branch (would risk data loss)"
-      exit 1
-    fi
-    INIT_HEAD=$(git rev-parse --abbrev-ref HEAD)
-    # Get all commits since {{ start }}, in chronological order
-    while read -r commit; do
-      git -c advice.detachedHead=false checkout "${commit}" || exit 1
-      { just debug_justfile={{ debug_justfile }} cargo +{{ rust }} build --locked --profile=dev --target=x86_64-unknown-linux-gnu; } || exit 1
-    done < <(git rev-list --reverse "{{ start }}".."$(git rev-parse HEAD)")
-    # Return to the initial branch if any (exit "detached HEAD" state)
-    git checkout "${INIT_HEAD}"
+    git rebase --keep-base "{{ start }}" --exec "{{ command }}"
```

Comment:
I'm a bit wary of side effects from the rebase here. I didn't check, but I'd expect it to change the commit date and the SHAs of the commits? At the very least, I think we need `--no-autosquash`.

What's the motivation for replacing this, if it's been working so far?

Reply:
I have never seen it change the shas.  The `--no-autosquash` is a good idea tho.

The motivation was just to simplify the code (basically, to push maintenance out to git where they already maintain the same logic).

I'm not devoted to the change.  It just seems like a simple cleanup.

Reply:
OK, if it mostly doesn't change the commit then I guess it's fine. I'm not strongly opposed to it, less maintenance makes sense, too. Let's add `--no-autosquash` and go for it in that case

Reply:
(I've got `autosquash` enabled by default in my config, which is usually what I want, but I would like the `build sweep` to squash my commits.)